### DISCUSSION TITLE
Can the defense phase in Simulating Invasion be changed

### DISCUSSION THREAD

**Discussion by MJL-JLU** [PMEAL/OpenPNM#2870] [Q&A]
Dear professor, may I ask if the Simulating Invasion example in openpnm default to the defense phase is water? Can I use the same method to simulate the phenomenon of water invading gas in hydrophobic porous media, or simulate the invasion of non-wetting phase fluids into another non-wetting phase fluid?

**Comment by jgostick** [comment]
The invasion algs in OpenPNM are pretty simple and actually the defending phase is irrelevant.  You just specify the properties of the system for the invading phase, so it can be water invading to displace air, or whatever.

**Comment by MJL-JLU** [comment]
I see. Thank you very much for your answer, Professor.

**Comment by MJL-JLU** [comment]
> The invasion algs in OpenPNM are pretty simple and actually the defending phase is irrelevant. You just specify the properties of the system for the invading phase, so it can be water invading to displace air, or whatever.

Professor, I also have another question. In the simulation of predicting relative permeability, the contact angle of the fluid refers to the contact angle between the fluid and the solid phase, air, or the contact angle between the fluid and the other phase fluid, solid phase. If one phase of the fluid is gas, how to give the parameters ? The same problem also exists for the addition of surface tension parameters.
